src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -813,4 +845,62 @@ RippleAddress RippleAddress::createSeedGeneric (std::string const& strText)
     return naNew;
 }
 
+uint256 keyFromSeed (uint128 const& seed)
+{
+    Serializer s;
+
+    s.add128 (seed);
+    uint256 result = s.getSHA512Half();
+
+    s.secureErase ();
+
+    return result;
+}
+
+RippleAddress getSeedFromRPC (Json::Value const& params)
+{
+    bool const has_passphrase = params.isMember (jss::passphrase);
+    bool const has_seed       = params.isMember (jss::seed);
+    bool const has_seed_hex   = params.isMember (jss::seed_hex);
+
+    int const n_secrets = has_passphrase + has_seed + has_seed_hex;
+
+    if (n_secrets > 1)
+    {
+        // `passphrase`, `seed`, and `seed_hex` are mutually exclusive.
+        return RippleAddress();
```

Comment:
@ximinez I don't think the `pair` is necessary. Rationale: `RippleAddress` is a deprecated legacy class that we want to replace. One of its deficiencies is that it encodes semantics for "invalid object." But since this function has no choice but to traffic in a `RippleAddress`, we might as well make use of its questionable feature. A future refactoring may replace `RippleAddress` with `Seed` in this code. In that case, we might consider this signature:
```
using JsonError = Json::Value;
std::pair<boost::optional<Seed>, JsonError>
getSeedFromRPC(Json::Value const& params)
```